### PR TITLE
Resolve for repeat

### PR DESCRIPTION
**Pull request by domoritz**
cc @willium

### CODE REVIEW

File: src/compile/repeat.ts [hunk 1]
```diff
@@ -127,44 +132,39 @@ export class RepeatModel extends Model {
...
-
-            // remove merged scales from children
-            delete child.component.scales[channel];
+      // Check whether the scales are actually compatible, e.g. use the same sort or throw error
+      keys(child.component.scales).forEach((channel: ScaleChannel) => {
+        if (this.resolve[channel].scale === 'shared') {
+          const childScale = child.component.scales[channel];
+          const modelScale = scaleComponent[channel];
+
+          if (!childScale || isSignalRefDomain(childScale.domain) || (modelScale && isSignalRefDomain(modelScale.domain))) {
```
Comment (kanitw) [inline src/compile/repeat.ts]: FYI, I don't stand the logic for this line.  I know you are not really changing anything here, but this TODO things seems worth noting as an issue that we should deal with before 2.0?

File: src/resolve.ts [hunk 2]
```diff
@@ -46,3 +46,11 @@ export function initLayerResolve(resolve: ResolveMapping): ResolveMapping {
 
   return out;
 }
+
+export function initLayerResolve(resolve: ResolveMapping): ResolveMapping {
+  return initResolve(resolve, channel => 'shared');
+}
+
+export function initRepeatResolve(resolve: ResolveMapping): ResolveMapping {
```
Comment (kanitw) [inline src/resolve.ts]: I thought in #2331 we agree not to do initResolve?

Comment (domoritz) [inline src/resolve.ts]: Yeah, but that can come later. Basically, I will add an argument to pass in the channels.

File: src/compile/repeat.ts (hunk 1 above)
Comment (domoritz) [inline src/compile/repeat.ts]: Ok

File: src/resolve.ts (hunk 2 above)
Comment (kanitw) [inline src/resolve.ts]: Ok

**Comment by kanitw** [comment]
Feel free to merging after filing an issue.

**Pull request merged**